cli/lsp/backend.rs

```diff
@@ -0,0 +1,905 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::anyhow;
+use deno_core::error::AnyError;
+use deno_core::serde_json;
+use deno_core::serde_json::json;
+use deno_core::serde_json::Value;
+use deno_core::ModuleSpecifier;
+use dprint_plugin_typescript as dprint;
+use lspower::jsonrpc::Error as LSPError;
+use lspower::jsonrpc::Result as LSPResult;
+use lspower::lsp_types::*;
+use lspower::Client;
+use lspower::LanguageServer;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::sync::RwLock;
+use tokio::fs;
+
+use crate::import_map::ImportMap;
+use crate::media_type::MediaType;
+use crate::tsc_config::TsConfig;
+
+use super::analysis;
+use super::capabilities;
+use super::config::Config;
+use super::diagnostics;
+use super::diagnostics::DiagnosticCollection;
+use super::diagnostics::DiagnosticSource;
+use super::lsp_extensions::VirtualTextDocumentParams;
+use super::memory_cache::MemoryCache;
+use super::sources::Sources;
+use super::text;
+use super::text::apply_content_changes;
+use super::tsc;
+use super::tsc::TsServer;
+use super::utils;
+
+#[derive(Debug, Clone)]
+pub struct Backend {
```

Comment:
nit: I really hate the term "backend"... `LanguageServer`?

Reply:
or `DenoLanguageServer` so as to not conflict with the trait?

Reply:
Renamed to `DenoLanguageServer`